src/docfx/build/page/BuildPage.cs

```diff
@@ -18,25 +19,26 @@ internal static class BuildPage
         {
             Debug.Assert(file.ContentType == ContentType.Page);
 
-            var (errors, isPage, outputModel, (inputMetadata, metadataObject)) = await Load(context, file);
-            var (generateErrors, outputMetadata) = await GenerateOutputMetadata(context, file, tocMap, inputMetadata, outputModel);
+            var (errors, isPage, (pageMetadata, pageModel), (inputMetadata, metadataObject)) = await Load(context, file);
+            var (generateErrors, outputMetadata) = await GenerateOutputMetadata(context, file, tocMap, inputMetadata, pageMetadata);
             errors.AddRange(generateErrors);
 
-            var mergedOutputMetadata = new JObject();
-            JsonUtility.Merge(mergedOutputMetadata, metadataObject);
-            JsonUtility.Merge(mergedOutputMetadata, JsonUtility.ToJObject(outputMetadata));
             var outputPath = file.GetOutputPath(outputMetadata.Monikers, file.Docset.SiteBasePath, isPage);
 
             object output = null;
             JObject metadata = null;
             if (isPage)
             {
-                (output, metadata) = ApplyPageTemplate(context, file, mergedOutputMetadata, outputMetadata.Conceptual);
+                var mergedOutputModel = new JObject();
+                JsonUtility.Merge(mergedOutputModel, metadataObject);
```

Comment:
How about adding a new method for this usage:
```csharp
        public static void Merge(JObject container, params JObject[] overwrites)
        {
            foreach (var item in overwrites)
            {
                Merge(container, item);
            }
        }
```

Reply:
we can add it later once we need apply data page template and need merge there.